projects/graphql-client/src/model/graphql-argument.ts

```diff
@@ -23,3 +25,51 @@ export type GraphQlArgumentValue =
   | GraphQlArgumentObject
   | GraphQlArgumentArray
   | GraphQlEnumArgument<string>;
+
+/**
+ * @param value
+ * Convert the value to GraphQL argument value
+ */
+export const getValueAsGraphQlArgument = (
+  value: unknown,
+  considerAllCapsStringAsEnum: boolean = false,
```

Comment:
I am not sure if we should do this. This makes it hard for us to combine enum and non enum cases that might use upper case input.
I'd prefer to have a boolean indicating whether the value is enum or not.

Reply:
Yeah, this would need more iterations in the future. The challenge is coming due to the nested nature of the object. I would like us to consider this as the first solution towards a better/generic solution. we can't have a boolean for each value.

Reply:
Yes, we can't have boolean for each value, but the only value that needs special treatment is enum. A string remains a string or becomes an enum. I feel its better to make the input param represent the type than use `considerAllCapsStringAsEnum`. That way we can use it for enums that aren't uppercased and strings that need to be uppercased without confusion.

Reply:
Yeah, that's what I tried initially, but I am not sure how to differentiate between a string and an enum. `typeof` wouldn't work.

Reply:
happy to address it (if there is a solution) in a follow up pr